src/pgduckdb_table_am.cpp

```diff
@@ -492,9 +495,35 @@ duckdb_am_handler(FunctionCallInfo /*funcinfo*/) {
 }
 }
 
+static duckdb::unordered_map<const TableAmRoutine * /*am*/, duckdb::string /*name*/> duckdb_table_ams = {
+    {&duckdb_methods, "duckdb"}};
+
+extern "C" __attribute__((visibility("default"))) void
+RegisterDuckdbTableAm(const char *name, const TableAmRoutine *am) {
+	duckdb_table_ams[am] = name;
```

Comment:
Let's check first if it's already in there, so extensions cannot accidentally override already existing ones.